tezster.js

```diff
@@ -163,32 +164,42 @@ program
 /*******deploy contract written in Michelson*/
 program
     .command('deploy')
-    .usage(`\n(This command provides the interactive shell)`)
+    .option('-i', '--interactive', 'use interactive mode')
+    .usage(`\n(This command provides the interactive shell to deploy a smart contract)`)
     .description('Deploys a smart contract written in Michelson')
     .action(function() {
-        if (process.argv.length > 3) {
-            console.log('Incorrect usage of deploy command. Correct usage: - tezster deploy');
+        if(process.argv[3] != '-i' && process.argv.length > 6) {
+            tezstermanager.deployContractNonInterativeMode();
             return;
+        } else if(process.argv[3] === '-i' && process.argv.length < 6) {
+            prompt(contractDeployParameters).then(deployParamaterValues => {
+                tezstermanager.deployContractInterativeMode(deployParamaterValues);
+            });
+            return;
+        } else {
+            console.log('Incorrect usage of deploy command. Correct usage:- tezster deploy <contract-label> <contract-file-path> <initial-storage> <account> <--optional amount>\nTo run command in interactive mode:- tezster deploy -i');
         }
-        prompt(contractDeployParameters).then(deployParamaterValues => {
-            tezstermanager.deployContract(deployParamaterValues);
-        });
 });
 
 
 /*******calls contract written in Michelson*/
 program
     .command('call')
-    .usage(`\n(This command provides the interactive shell)`)
+    .option('-i', '--interactive', 'use interactive mode')
+    .usage(`\n(This command provides the interactive shell to call deployed smart contract)`)
     .description('Calls a smart contract with given value provided in Michelson format')
     .action(function() {
-        if (process.argv.length > 3) {
-            console.log('Incorrect usage of call command. Correct usage: - tezster call');
+        if(process.argv[3] != '-i' && process.argv.length > 5) {
+            tezstermanager.callContractNonInterativeMode();
             return;
+        } else if(process.argv[3] === '-i' && process.argv.length < 5) {
```

Comment:
if user has provided -i flag, you can why do we need to check for argv.length being less than 5?

Reply:
I am implementing the condition, if there is flag "-i", it won't work for extra arguments after flag. For ex. "tezster deploy -i" will provide interactive shell, while "tezster deploy -i <any-arg>" will throw error. For non-interactive user have to provide arguments without "-i" flag.

Reply:
I get that. That's why i'm telling that checing 'process.argv[3] === '-i'' is sufficient/ 
if args length is more than 6, it'll fail 
you can try with 'tezster call -i arg1 agr2 arg3 arg4 arg5 arg6 arg7"

Reply:
You logic should check if there's -i or not as 3rd argument and proceed accrodingly. 
checking  "process.argv[3] !== '-i' && process.argv.length > 6" is also very fragile way of doing it.